cloudflare/resource_cloudflare_page_rule_test.go

```diff
@@ -279,6 +279,24 @@ func TestTranformForwardingURL(t *testing.T) {
 	}
 }
 
+// This test ensures there is no crash while encountering a nil query_string section, which may happen when updating
+// existing Page Rule that didn't have this value set previously
+func TestCacheKeyFieldsNilValue(t *testing.T) {
+	pageRuleAction, err := transformToCloudflarePageRuleAction(
+		"cache_key_fields",
+		[]interface{}{map[string]interface{}{"cookie": []interface{}{map[string]interface{}{"check_presence": []interface{}{}, "include": []interface{}{"next-i18next"}}}, "header": []interface{}{map[string]interface{}{"check_presence": []interface{}{}, "exclude": []interface{}{}, "include": []interface{}{"x-forwarded-host"}}}, "host": []interface{}{map[string]interface{}{"resolved": false}}, "query_string": []interface{}{interface{}(nil)}, "user": []interface{}{map[string]interface{}{"device_type": true, "geo": true, "lang": true}}}},
```

Comment:
types, amirite? 😅

Reply:
This is the exact setting that triggers crash for the used who reported it. Put here as a regression test.